### ISSUE TITLE
Secret key appears to not spawn occasionally

### ISSUE THREAD

**Issue by SheerSt** [SheerSt/pokemon-wilds#3]
Currently looking into.

**Comment by SheerSt** [comment]
@odd1sh can you upload your save file here?

**Comment by odd1sh** [comment]
> @odd1sh can you upload your save file here?

[minecraft - Copy.zip](https://github.com/SheerSt/pokemon-wilds/files/5264175/minecraft.-.Copy.zip)

**Comment by SheerSt** [comment]
@odd1sh okay, so I looked into it and the key spawned on your map, but it's in an almost impossible place to find. It's right above the building, behind a tree. I have it circled in red:
![wow](https://user-images.githubusercontent.com/8891626/93959376-27650700-fd1e-11ea-85ac-91ccda265abb.png)

While that's terribly unlucky, I feel terrible and I'm really sorry this happened. It seems like you've actually put some time into the game too and that makes me feel even worse (you have like a level 58 larvitar lol). Seriously don't feel obligated to keep playing or anything... the fact that you found this issue has been a tremendous help to me and I'm very grateful.

I will probably just remove the secret key requirement from entering the building. I didn't think about how hard it would be to find on large maps, especially if it has a chance of spawning in such a troll location. I'm open to suggestions.

I attached a save file where I walked over and picked up the secret key, and then walked back (attached). Feel free to use it if you desire (you will need to rename from .zip to .sav)
[minecraft-fixed.zip](https://github.com/SheerSt/pokemon-wilds/files/5265304/minecraft-fixed.zip)

**Comment by odd1sh** [comment]
Thank you so much, I've been looking for that key for ages! Also don't feel bad because I didn't really put THAT much time into the game, I just caught a Stantler, jumped up onto the mountain biomes and kept spamming pokeballs at the high leveled mons there. That's why my Larvitar is level 58 because I just recently caught it :P. But thank you so much for this and I can't wait to see how the game turns out in the future!

**Comment by odd1sh** [comment]
@SheerSt 

Hey, I just finished the dungeon, but I'm stuck. After you complete the lowest level and try to climb back to the top, the walls that are activated by the switch are in your way again, I can't reactivate them because the switch for them is on the upper floor, and they're also invisible...so yeah, I'm permanently stuck in the dungeon. ):

**Comment by SheerSt** [comment]
@odd1sh hmm that shouldn't happen... I'll look again at your save file and see what's up. Once you flip a secret switch, the door should stay permanently in their flipped position, going up and down stairs shouldn't change that. Also, as a fallback you can always just faint all of your Pokemon, and youll respawn outside (i think)

**Comment by SheerSt** [comment]
@odd1sh for some reason the pokemon mansion building in your save file is super glitched. When you walk into the building, the music is supposed to switch and the light should dim - neither of those happen in your file. Also, there are random invisible walls, I have no idea how that's possible. Also, you should be able to encounter wild pokemon, which I don't see in your file (so therefore fainting your pokemon in order to soft reset won't work). I'm going to have to look into it further, it's not obvious yet what the issue is.

**Comment by SheerSt** [comment]
@odd1sh The issue was on my end, the game wasn't saving right. I have updated the exe under builds/latest with a new version that fixes the issues. I temporarily upped the catch rate of Mewtwo to try to make him easier to catch. As it is now, I noticed that it's almost impossible to catch him.

While doing this I encountered some other issues that need to be fixed:
 - need to be able to transfer pokemon once you catch them
 - player will attempt to interact with object in front of them as soon as game is loaded
 - fainted pokemon will sometimes be sent out at the start of a battle.
 - mewtwo battle music stops when map fades to day

**Comment by odd1sh** [comment]
> @odd1sh The issue was on my end, the game wasn't saving right. I have updated the exe under builds/latest with a new version that fixes the issues. I temporarily upped the catch rate of Mewtwo to try to make him easier to catch. As it is now, I noticed that it's almost impossible to catch him.
> 
> While doing this I encountered some other issues that need to be fixed:
> 
> * need to be able to transfer pokemon once you catch them
> * player will attempt to interact with object in front of them as soon as game is loaded

So should I redownload the game or something?

**Comment by SheerSt** [comment]
@odd1sh yeah you will need to redownload the exe if you if you want to use thr updated version
